packages/manager/apps/web/client/app/hosting/cdn/shared/hosting-cdn-shared-settings.controller.js

```diff
@@ -0,0 +1,343 @@
+import set from 'lodash/set';
+import get from 'lodash/get';
+import find from 'lodash/find';
+import forEach from 'lodash/forEach';
+import assign from 'lodash/assign';
+import filter from 'lodash/filter';
+import maxBy from 'lodash/maxBy';
+import concat from 'lodash/concat';
+import map from 'lodash/map';
+import clone from 'lodash/clone';
+import includes from 'lodash/includes';
+
+import {
+  HOSTING_CDN_SHARED_SETTINGS_RULE_CACHE_RULE,
+  HOSTING_CDN_SHARED_SETTINGS_RULE_FACTOR_DAY,
+  HOSTING_CDN_SHARED_SETTINGS_RULE_FACTOR_HOUR,
+  HOSTING_CDN_SHARED_SETTINGS_RULE_FACTOR_MINUTE,
+  HOSTING_CDN_SHARED_SETTINGS_RULE_TYPE_BROTLI,
+  HOSTING_CDN_SHARED_SETTINGS_RULE_TYPE_DEVMODE,
+} from './hosting-cdn-shared-settings.constants';
+
+export default class CdnSharedSettingsController {
+  /* @ngInject */
+  constructor(
+    $q,
+    $scope,
+    $state,
+    $translate,
+    $uibModal,
+    Alerter,
+    HostingCdnSharedService,
+  ) {
+    this.$q = $q;
+    this.$scope = $scope;
+    this.$state = $state;
+    this.$translate = $translate;
+    this.Alerter = Alerter;
+    this.HostingCdnSharedService = HostingCdnSharedService;
+  }
+
+  $onInit() {
+    this.loading = { init: false };
+    this.params = {
+      domain: this.domain,
+      domainName: this.domainName,
+      serviceName: this.serviceName,
+    };
+    this.model = {
+      alwaysOnline: {
+        enabled: true,
+        changed: false,
+      },
+      http: {
+        enabled: true,
+        changed: false,
+      },
+      devmode: null,
+      brotli: null,
+      rules: [],
+    };
+    this.tasks = {
+      toAdd: [],
+      toUpdate: [],
+      toRemove: [],
+    };
+    this.HostingCdnSharedService.model = this.model;
+    this.HostingCdnSharedService.tasks = this.tasks;
+
+    if (!this.params.domain) {
+      this.goBack();
+    }
+
+    this.HostingCdnSharedService.getCDNDomainsDetails(
+      this.params.serviceName,
+      this.params.domainName,
+    ).then((options) => {
+      this.model.rules = filter(options, {
+        type: HOSTING_CDN_SHARED_SETTINGS_RULE_CACHE_RULE,
+      });
+      this.model.devmode = assign(
+        find(options, { type: HOSTING_CDN_SHARED_SETTINGS_RULE_TYPE_DEVMODE }),
+        { changed: false },
+      );
+      this.model.brotli = assign(
+        find(options, { type: HOSTING_CDN_SHARED_SETTINGS_RULE_TYPE_BROTLI }),
+        { changed: false },
+      );
+      this.model.rulesName = map(this.model.rules, (r) => r.name);
+      this.loading.init = true;
+    });
+  }
+
+  getSwitchBtnStatusText(switchBtn) {
+    return this.$translate.instant(
+      `hosting_cdn_shared_state_${switchBtn ? 'enable' : 'disabled'}`,
+    );
+  }
+
+  convertToUnitTime(ttl) {
```

Comment:
Can't moment handle this conversion ? Especially it seems duplicated

Reply:
i dont need moment for that.
I juste need to % and divide to compute initial user unit time value.
because the value has been stored as sec. from API side.